tests/index.js

```diff
@@ -27,8 +27,8 @@ describe('classNames', function () {
 		assert.equal(classNames('', 'b', {}, ''), 'b');
 	});
 
-	it('returns an empty string for an empty configuration', function () {
-		assert.equal(classNames({}), '');
+	it('returns undefined for an empty configuration', function () {
```

Comment:
you should check if undefined will become string in the template

Reply:
empty string is better and predictable for UI developers.

Reply:
... the whole point of this PR is to make it undefined. Read the above posts to understand why.